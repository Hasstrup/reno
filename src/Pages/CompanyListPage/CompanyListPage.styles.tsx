import { createUseStyles } from 'react-jss';

export const CompanyListPageStyles = createUseStyles({
    ListItemContainer: {
        padding: '0 !important'
    }
})