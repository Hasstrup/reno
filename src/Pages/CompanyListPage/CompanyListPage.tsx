import React, { FC, useState } from "react";
import { List, Typography } from "antd";
import { CompanyListPageProps } from "./CompanyListPage.types";
import { CompanyListPageStyles as useStyles } from "./CompanyListPage.styles"
import { CompanyListItem } from "@/Pages/CompanyListPage/Components/CompanyListItem/CompanyListItem";
import { ListItem } from "@/Pages/CompanyListPage/Components/CompanyListItem/CompanyListItem.types"
import store from "./Store/store.json";


export const CompanyListPage: FC<CompanyListPageProps> = (props) => {
  const classes = useStyles()
  const source = JSON.parse(JSON.stringify(store)).data as ListItem[]
  console.log(source)
  const [dataSource, setDataSource] = useState(source)

  return (
    <List
      grid={{ gutter: 1, column: 4 }}
      header={<Typography>{`Add new criteria`}</Typography>}
      dataSource={dataSource}
      bordered
      renderItem={(item: ListItem) => {
        return (
          <List.Item className={classes.ListItemContainer}>
            <CompanyListItem item={item} />
          </List.Item>
        );
      }}
    />
  );
};
