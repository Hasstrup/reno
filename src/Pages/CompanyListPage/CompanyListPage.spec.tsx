import React from "react";
import { CompanyListPage } from "./CompanyListPage";
import { shallow, mount } from "enzyme";

describe("<CompanyListPage />", () => {
  let wrapper: any;
  describe("Context: Shallow rendering", () => {
    beforeEach(() => {
      wrapper = shallow(<CompanyListPage />);
    });

    it("renders correctly", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Context: Deep rendering", () => {
    beforeEach(() => {
      Object.defineProperty(window, "matchMedia", {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
          matches: false,
          media: query,
          onchange: null,
          addListener: jest.fn(),
          removeListener: jest.fn(),
          addEventListener: jest.fn(),
          removeEventListener: jest.fn(),
          dispatchEvent: jest.fn(),
        })),
      });

      wrapper = mount(<CompanyListPage />);
    });

    it("adds a new criteria when the add criteria button is clicked", () => {
       expect(wrapper.find('LegendLineItem')).toHaveLength(6)
       wrapper.find("button.add__new__criteria").simulate('click')
       expect(wrapper.find('LegendLineItem')).toHaveLength(7)
    });
  });
});
