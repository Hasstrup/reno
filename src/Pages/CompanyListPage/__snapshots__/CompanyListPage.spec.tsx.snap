// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyListPage /> Context: Shallow rendering renders correctly 1`] = `ShallowWrapper {}`;
